src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -121,3 +116,26 @@ def _should_store_transaction_name(event_data: Mapping[str, Any]) -> bool:
         return False
 
     return True
+
+
+def _bump_rule_lifetime(project: Project, event_data: Mapping[str, Any]) -> None:
+    from sentry.ingest.transaction_clusterer import rules as clusterer_rules
+
+    applied_rules = event_data.get("_meta", {}).get("transaction", {}).get("", {}).get("rem", {})
+    if not applied_rules:
+        return
+
+    stored_rules = clusterer_rules.get_redis_rules(project)
```

Comment:
Not returning early when there aren't any rules because I believe it should not be the case in most cases, so overall less benefit of skipping the (short) loop below. I don't have data to back this up though.